diff --git a/dist/vite/node-adapter.js b/dist/vite/node-adapter.js
index e5ef9363d5eedf94d5a13ef53fc8ca9893547130..d2d9f4882df4e08877c709b886a1208b0c5d75d5 100644
--- a/dist/vite/node-adapter.js
+++ b/dist/vite/node-adapter.js
@@ -36,7 +36,7 @@ function fromNodeHeaders(nodeHeaders) {
 
 // Based on `createRemixRequest` in packages/remix-express/server.ts
 function fromNodeRequest(nodeReq, nodeRes) {
-  let origin = nodeReq.headers.origin && "null" !== nodeReq.headers.origin ? nodeReq.headers.origin : `http://${nodeReq.headers.host}`;
+  let origin = `https://${nodeReq.headers.host}`;
   // Use `req.originalUrl` so Remix is aware of the full path
   invariant["default"](nodeReq.originalUrl, "Expected `nodeReq.originalUrl` to be defined");
   let url = new URL(nodeReq.originalUrl, origin);
