diff --git a/dist/node/chunks/node.js b/dist/node/chunks/node.js
index 21f8d7cf6b0c9e9f473b891b9b1de0d7f857179b..8a1ce7a4fa37682dcb823bdf78ec987d8d98c914 100644
--- a/dist/node/chunks/node.js
+++ b/dist/node/chunks/node.js
@@ -8567,0 +8568 @@ var require_connect = /* @__PURE__ */ __commonJSMin(((exports, module) => {
+		req$4.url ??= "/";
@@ -8576,0 +8578 @@ var require_connect = /* @__PURE__ */ __commonJSMin(((exports, module) => {
+			req$4.url ??= req$4.originalUrl;
