diff --git a/src/scroll-timeline-base.js b/src/scroll-timeline-base.js
index 2c854928701c5dae23c985eb04f73a6ca2b661c3..abf1a7629568f411dd83dd509640850706d59528 100644
--- a/src/scroll-timeline-base.js
+++ b/src/scroll-timeline-base.js
@@ -162,13 +162,19 @@ function isValidAxis(axis) {
  */
 export function measureSource (source) {
   const style = getComputedStyle(source);
+  const clientHeight = source.clientHeight;
   return {
     scrollLeft: source.scrollLeft,
     scrollTop: source.scrollTop,
     scrollWidth: source.scrollWidth,
     scrollHeight: source.scrollHeight,
     clientWidth: source.clientWidth,
-    clientHeight: source.clientHeight,
+    get clientHeight() {
+      if (document.scrollingElement === source) {
+        return window.innerHeight;
+      }
+      return clientHeight;
+    },
     writingMode: style.writingMode,
     direction: style.direction,
     scrollPaddingTop: style.scrollPaddingTop,
